### PR TITLE
Update app-actions.md

### PR DESCRIPTION
**Pull request by theexperimenter**
I have updated the docs to remove the console.log() statement as it was confusing our clients. 

Also added in a section for testing and debugging app actions as lots of people have been confused on how to actually test and debug action.

### CODE REVIEW

File: docs/API/core/app-actions.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ To start, use the `Fliplet.Page.onRemoteExecution()` function to register your c
 Fliplet.Page.onRemoteExecution(function (payload) {
   // This code will run when the screen is triggered by an app action.
 
-  console.log('App action sent this payload', payload);
+  // Here you can access the payload sent to the action when it was called.
```
Comment (squallstar) [inline docs/API/core/app-actions.md:51]: I would prefer keeping the console log as it makes clear the variable can be used. Can you confirm why clients found it confusing?

File: docs/API/core/app-actions.md [hunk 2]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
 
 ---
 
+## Testing and debugging an action
+
+Once you have created an app action and added the code you want to run to the onRemoteExecution() function you should test it and see if it running as expected. Note that you cannot use developer tools to view your console.log inside the onRemoteExecution() function as this is running on the server side. Here is the general steps to test an app action:
```
Comment (squallstar) [inline docs/API/core/app-actions.md:331]: Wrap function names and variables with these carets:
```
`onRemoteExecution()`
```

File: docs/API/core/app-actions.md [hunk 3]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
...
 ---
 
+## Testing and debugging an action
+
+Once you have created an app action and added the code you want to run to the onRemoteExecution() function you should test it and see if it running as expected. Note that you cannot use developer tools to view your console.log inside the onRemoteExecution() function as this is running on the server side. Here is the general steps to test an app action: 
+
+1. Create the app action 
+2. Add code to onRemoteExecution() that you want to run in the app action. 
+3. Call the app action on demand and log the returned data to see if it working as expected. 
+4. If there are multiple steps in your code then move the `return Promise.resolve()` to the point where you want to see the data available.
```
Comment (squallstar) [inline docs/API/core/app-actions.md:336]: I don't understand what this is supposed to do

File: docs/API/core/app-actions.md [hunk 4]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
...
+
+1. Create the app action 
+2. Add code to onRemoteExecution() that you want to run in the app action. 
+3. Call the app action on demand and log the returned data to see if it working as expected. 
+4. If there are multiple steps in your code then move the `return Promise.resolve()` to the point where you want to see the data available. 
+
+```js
+// If an app action called 'test-action' exists then we can call in on another screen
+Fliplet.App.Actions.runWithResult('test-action').then(function (result) {
+  // If our app action working then it should return the entries from the data source query
```
Comment (squallstar) [inline docs/API/core/app-actions.md:341]: [...] "is" working "correctly" [...]

File: docs/API/core/app-actions.md [hunk 5]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
...
+  console.log(result)
```
Comment (squallstar) [inline docs/API/core/app-actions.md:342]: Missing semicolon

File: docs/API/core/app-actions.md [hunk 6]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
...
+});
+
+//This code exists on another screen and queries the data source and returns the data
```
Comment (squallstar) [inline docs/API/core/app-actions.md:345]: Empty space after `//` for consistency

File: docs/API/core/app-actions.md [hunk 7]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
...
+    .then(function (connection) {
+      return connection.find({
+        where: {
+            Name: 'John'
+        }
+      });
+    })
+    .then(function (records) {
+        //In order to see the data currently we need to return it in promise so that we can log in when the action is called above. 
+        return Promise.resolve(records)
```
Comment (squallstar) [inline docs/API/core/app-actions.md:357]: This part is not necessary, the promise above already returns `records`

File: docs/API/core/app-actions.md [hunk 8]
```diff
@@ -326,6 +326,42 @@ Fliplet.App.Actions.remove('confirm-booking').then(function () {
...
+    });
+});
+
+```
+
+Note: You have to ensure that if you are executing asynchronous javaScript then the promises are chained correctly. If you do not chain the promises correctly then it is likely that no payload will be returned.
```
Comment (squallstar) [inline docs/API/core/app-actions.md:363]: "JavaScript".

This however applies to all our JS API, not just App Actions so I don't think it should be here. @tonytlwu thoughts?

**Review by squallstar — CHANGES_REQUESTED** [review]

File: docs/API/core/app-actions.md (hunk 1 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:51]: So clients would just copy this snippet and are confused why the console never runs. It does not run on the page it was pasted in and neither does this run when calling the action on demand. Its doesn't seem like there is any way to make this console run. Does that make sense?

File: docs/API/core/app-actions.md (hunk 3 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:336]: Users are not sure how to test their app action code. This is general steps to debug step by step. People are creating and setting up the actions but have no idea to verify it is working as expected.

File: docs/API/core/app-actions.md (hunk 8 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:363]: I agree but I have found that app actions are a lot less forgiving and if you chain incorrectly it just will not run properly. It's a very common mistake and I just want to highlight here for people so if they find something is not running correctly they review their promises.

File: docs/API/core/app-actions.md (hunk 1 above)
Comment (squallstar) [reply docs/API/core/app-actions.md:51]: Ah I see, that makes sense. I'm good with removing the console log then.

File: docs/API/core/app-actions.md (hunk 2 above)
Comment (squallstar) [reply docs/API/core/app-actions.md:331]: @theexperimenter please resolve conversations when new commits have been pushed to address the comments.

Comment (theexperimenter) [reply docs/API/core/app-actions.md:331]: Agree with this change.

File: docs/API/core/app-actions.md (hunk 4 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:341]: Agreed.

File: docs/API/core/app-actions.md (hunk 5 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:342]: Agreed.

File: docs/API/core/app-actions.md (hunk 6 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:345]: Agreed.

File: docs/API/core/app-actions.md (hunk 7 above)
Comment (theexperimenter) [reply docs/API/core/app-actions.md:357]: Fair enough

**Pull request merged**